viya4-iac-aws-darksite/README.md

```diff
@@ -0,0 +1,39 @@
+
+### Procedures
+
+1.  **Build a Private VPC for EKS:**
+
+    - **Note:** _Creating AWS VPCs for an EKS cluster can be accomplished in many different ways. Some of those methods include using a CloudFormation template, the AWS console, the AWS CLI, or Terraform scripts._
+
+    - **Note**: If you have an existing VPC environment consisting of a private VPC and a public VPC (optional) that constitutes a Dark Site configuration, skip to the **_Create Custom AMI for Jumpserver/NFSServer_** in step #2 below.
+
+    - Background Information - AWS Reference documentation with details on how to create a VPC for any EKS cluster: https://docs.aws.amazon.com/eks/latest/userguide/creating-a-vpc.html
+
+    - For the cluster VPC ranges, `viya4-iac-aws` defaults to using a CIDR of /16. The `viya4-iac-aws` project creates a /21 VPC with both /22 and /28 sized subnets.  Excluding the `control_plane` subnets, the sizes were chosen by estimating the number of pods (doubling that number to account for viya updates), services, AWS overhead, and then adding a few hundred as a buffer.  
+
+2.  **Create Custom AMI for Jumpserver/NFSServer**
+    - The standard base AMI image used by `viya4-iac-aws` does not include the required NFS related Linux distributution packages.  Normally, `viya4-iac-aws` will attempt to install the NFS packages as part of the VM initialization.  In a Dark Site without access to Internet based resources, installation from an Internet based repository will not be possible.  To mitigate that issue, we'll need to create a custom AMI and then modify the local copy of our `viya4-iac-aws` repository to add references to that custom AMI as well as remove some of the initialization steps in the cloud-init files (for jumpserver and nfs-server). 
+
+    - Instructions to complete the custom AMI creation steps can be found [here](https://github.com/sassoftware/viya4-iac-aws/tree/feat/iac-1117/viya4-iac-aws-darksite/custom-ami/) [here](https://github.com/sassoftware/viya4-iac-aws/tree/main/viya4-aws-darksite/custom-ami/).
+
+3.  **Build tfvars:**
+    - The Terraform scripts create the necessary BYON subnet_ids, `viya4-iac-aws` can be configured to use those subnets for all the subnet_ids map BYON variables in the tfvars file (public, private, control_plane and database).
+    - Do not specify any public access cidrs in your tfvars file.
+    - Set `cluster_api_mode` to private.
```

Comment:
Technically, cluster_api_mode is not required to be at 'private' for a dark site.  IaC creates a 'public and private' endpoint when set to public and EKS still sets up the ENIs for private access to the control plane.  The primary thing that IaC users must ensure for a private site is a route to AWS services needed to create the cluster, over a private channel, by ensuring that either:
IaC creates VPC endpoints for requisite services, e.g. *.ec2.<region>.vpce.amazonaws.com
or the customer pre-creates VPC endpoints for requisite services

If one uses public things are still workable and in such a case a customer might want to specify cluster_endpoint_public_access_cidrs

Can we re-phrase this all as a recommended way to use IaC to integrated with a BYON dark site?

Recommend using cluster_api_mode of private, recommend ensuring vpc_private_endpoints_enabled is true and include all source IPs as needed for the following variables:

cluster_endpoint_private_access_cidrs
vpc_endpoint_private_access_cidrs
vm_private_access_cidrs